pkg/apis/core/v1alpha1/types_shoot.go

```diff
@@ -383,120 +383,120 @@ type ServiceAccountConfig struct {
 	// identifier in "iss" claim of issued tokens. This value is a string or URI.
 	// Defaults to URI of the API server.
 	// +optional
-	Issuer *string `json:"issuer,omitempty"`
+	Issuer *string `json:"issuer,omitempty" protobuf:"bytes,1,opt,name=issuer"`
 	// SigningKeySecret is a reference to a secret that contains an optional private key of the
 	// service account token issuer. The issuer will sign issued ID tokens with this private key.
 	// Only useful if service account tokens are also issued by another external system.
 	// +optional
-	SigningKeySecret *corev1.LocalObjectReference `json:"signingKeySecretName,omitempty"`
+	SigningKeySecret *corev1.LocalObjectReference `json:"signingKeySecretName,omitempty" protobuf:"bytes,2,opt,name=signingKeySecretName"`
```

Comment:
Again, sometimes we have `castType`, sometimes we don't. What's the policy?

Reply:
I am not adding the protobuf tags manually, they are added automatically by the generator. But I will try to check what is the rationale behind?

Reply:
Ah okay, even better. I was afraid that we have to maintain this manually... 😄

Reply:
`casttype` is used to change the generated field type. It allows to have proto message such as

```proto
// KubeProxyConfig contains configuration settings for the kube-proxy.
message KubeProxyConfig {
  optional KubernetesConfig kubernetesConfig = 1;

  // Mode specifies which proxy mode to use.
  // defaults to IPTables.
  // +optional
  optional string mode = 2;
}
```
where `mode` type is `string`.
The `mode` in the go type is of type `ProxyMode`

```golang
// KubeProxyConfig contains configuration settings for the kube-proxy.
type KubeProxyConfig struct {
	KubernetesConfig `json:",inline" protobuf:"bytes,1,opt,name=kubernetesConfig"`
	// Mode specifies which proxy mode to use.
	// defaults to IPTables.
	// +optional
	Mode *ProxyMode `json:"mode,omitempty" protobuf:"bytes,2,opt,name=mode,casttype=ProxyMode"`
}
```

and the generated unmarshal func in `generated.pb.go` properly casts to `ProxyMode`.

You can note that casttype is used for type aliases  - `ProxyMode`, `corev1.ServiceExternalTrafficPolicyType`, etc.